rpki-validator/src/main/java/net/ripe/rpki/validator3/domain/retrieval/TrustAnchorRetrievalService.java

```diff
@@ -0,0 +1,147 @@
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+import net.ripe.rpki.commons.validation.ValidationResult;
+import net.ripe.rpki.validator3.api.util.BuildInformation;
+import net.ripe.rpki.validator3.domain.ErrorCodes;
+import net.ripe.rpki.validator3.domain.metrics.HttpClientMetricsService;
+import net.ripe.rpki.validator3.domain.metrics.RsyncMetricsService;
+import net.ripe.rpki.validator3.util.Rsync;
+import net.ripe.rpki.validator3.util.RsyncFactory;
+import org.apache.commons.lang3.ArrayUtils;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.api.ContentResponse;
+import org.eclipse.jetty.http.HttpHeader;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URI;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeoutException;
+
+@Slf4j
+@Setter
+@Service
+public class TrustAnchorRetrievalService {
+    private boolean isFileProtocolEnabled = false;
+
+    @Value("${rpki.validator.rsync.local.storage.directory}")
+    private File localRsyncStorageDirectory;
+
+    @Autowired
+    private BuildInformation buildInformation;
+
+    @Autowired
+    private RsyncFactory rsyncFactory;
+
+    @Autowired
+    private HttpClient httpClient;
+
+    @Autowired
+    private HttpClientMetricsService httpMetrics;
+
+    @Autowired
+    private RsyncMetricsService rsyncMetrics;
+
+    public byte[] fetchTrustAnchorCertificate(URI trustAnchorCertificateURI, ValidationResult validationResult) throws IOException {
+        switch (trustAnchorCertificateURI.getScheme()) {
+            case "rsync":
+                return fetchRsyncTrustAnchorCertificate(trustAnchorCertificateURI, validationResult);
+            case "https":
+                return fetchHttpsTrustAnchorCertificate(trustAnchorCertificateURI, validationResult);
+            case "file":
+                if (isFileProtocolEnabled) {
+                    return Files.toByteArray(new File(trustAnchorCertificateURI));
+                }
+                validationResult.error(ErrorCodes.TRUST_ANCHOR_FETCH, trustAnchorCertificateURI.toASCIIString(), "File support not explicitly enabled.");
+                return null;
+            default:
+                validationResult.error(ErrorCodes.TRUST_ANCHOR_FETCH, trustAnchorCertificateURI.toASCIIString(), "Unsupported URI");
+                return null;
+        }
+    }
+
+    protected byte[] fetchHttpsTrustAnchorCertificate(URI trustAnchorCertificateURI, ValidationResult validationResult) throws IOException {
+        final long t0 = System.currentTimeMillis();
+        String statusDescription = "unknown";
+
+        try {
+            ContentResponse res = httpClient
+                    .newRequest(trustAnchorCertificateURI)
+                    .header(HttpHeader.USER_AGENT, null)
```

Comment:
Why ".header" two times?

Reply:
```java
    /**
     * <p>Adds the given {@code value} to the specified {@code header}.</p>
     * <p>Multiple calls with the same parameters will add multiple values;
     * use the value {@code null} to remove the header completely.</p>
     * ...
     */
```

Reply:
Turns out we have been setting the user agent in a strange fashion all the time, I'll refactor it to use a different method.